core/src/ast_builder/select/limit.rs

```diff
@@ -113,10 +121,17 @@ impl<'a> LimitNode<'a> {
 
 impl<'a> Prebuild for LimitNode<'a> {
     fn prebuild(self) -> Result<NodeData> {
-        let mut select_data = self.prev_node.prebuild()?;
-        select_data.limit = Some(self.expr.try_into()?);
+        let mut node_data = self.prev_node.prebuild()?;
+        match node_data {
+            NodeData::Select(ref mut select_data) => {
+                select_data.limit = Some(self.expr.try_into()?)
+            }
+            NodeData::Values(ref mut values_data) => {
+                values_data.limit = Some(self.expr.try_into()?)
+            }
```

Comment:
Is there better way to merge two cases? like
```rs
NodeData::Select(ref mut data) | NodeData::Values(ref mut data) => {
    data.limit = Some(self.expr.try_into()?)
}
```
which was failed cuz of type conflict though..